### PR TITLE
Automatic documentation generation

### PR DESCRIPTION
**Pull request by ViMaSter**
## Summary
This branch includes `/docs_generator`, which generates [`.MD`/Markdown-files](https://daringfireball.net/projects/markdown/) for the `ActionDecoder.cs`, identical to the current structure of our wiki.  
These are generated for every push and PR to `develop`, `releasecandidate/*` and `master` branch using GitHub Actions.

- Actor Actions ([wiki](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/wiki/Actor-Actions) vs. [generated](https://github.com/ViMaSter/GG-JointJustice-Unity/blob/develop/docs/methods/Actor.md))
- Preset Values ([wiki](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/wiki/Preset-Values) vs. [generated](https://github.com/ViMaSter/GG-JointJustice-Unity/blob/develop/docs/constants.md))

### Technical implementation
`/docs_generator` contains a .NET 6 C# Console App, which reads [`Assets/Scripts/TextDecoder/ActionDecoder.cs`](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/blob/f14af4e2bba25c64cb981eb07d41b2392de26632/Assets/Scripts/TextDecoder/ActionDecoder.cs) and generates markdown files based on the [xmldoc comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/) of `CAPITALZED_METHODS()`.
Files are generated for...
-  **every category** methods are assigned to, e.g.:
  https://github.com/ViMaSter/GG-JointJustice-Unity/blob/develop/unity/Assets/Scripts/TextDecoder/ActionDecoder.cs#L151
  https://github.com/ViMaSter/GG-JointJustice-Unity/blob/develop/unity/Assets/Scripts/TextDecoder/ActionDecoder.cs#L293

- and a `constants.md` containing **all files for `AssetNames`** (like sound effects, background images or actors), e.g.:
  https://github.com/ViMaSter/GG-JointJustice-Unity/blob/develop/unity/Assets/Scripts/TextDecoder/ActionDecoder.cs#L222
  https://github.com/ViMaSter/GG-JointJustice-Unity/blob/develop/unity/Assets/Scripts/TextDecoder/ActionDecoder.cs#L517

## Minor notices
- `CLEAR_SPEED()`, `WAIT_FOR_INPUT()`, `NARRATE()` exist in the Wiki, but don't exist in code.
- `RELOAD_SCENE()` has no Wiki entry.
- The actor file for for `JudgeBrent` was called `BrentJudge`. Updated it to `JudgeBrent`, to be consistent.
- **Qualified `AssetName`s even more explicitly**:
  - [The wiki has a page on "preset values"](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/wiki/Preset-Values), which lists which characters we have, their animations, our music, etc.
    In order for the generator to somehow figure out, which files are applicable where, I've split up the `AssetName` classes:
    ```cs
    /// <param name="evidence" validFiles="Assets/ScriptableObjects/Evidence/*.asset">Name of evidence</param>
    private void REMOVE_EVIDENCE(EvidenceAssetName evidence) { }
    ```
    In this case, `EvidenceAssetName` is mapped to all files matching the `Assets/ScriptableObjects/Evidence/*.asset` qualifier
    The new classes are:
     - FullscreenAnimationAssetName
     - SceneAssetName
     - SfxAssetName
     - SongAssetName
     - EvidenceAssetName
     - ActorPoseAssetName
     - ActorAssetName

## Before/after screenshots and/or animated gif
# ⚠️ MERGING THIS MOVES THE GAME! `/` -> `/unity`! ⚠️ 
This is how our repo will look after this merge: https://github.com/ViMaSter/GG-JointJustice-Unity
All GitHub Actions are updated accordingly. Also the game build is only triggered, if files inside this `unity` subfolder change.

The folders:
- `/unity`: contains everything related to the Unity project
- `/docs_generator`: contains a .NET 6 Console Application which generates `/docs`
- `/docs`: contains [`.MD`/Markdown-files](https://daringfireball.net/projects/markdown/) which are generated whenever the project inside `/docs_generator` is run  
  (currently on every PR to `develop`, `releasecandidate/*` and `master`)

## Testing instructions
Pull down this PR and make sure everything inside Unity still works.

After switching to this branch, your working directory might still contain previously untracked files:
image
Either...
- move them to `/unity` by hand (your Unity project should load them as normal)
- run `git clean -d -n` (this will **not** remove any files yet) to see which files are unused and **then** run `git clean -d -f`  
  (❗ this method **will delete** the previously listed files and Unity will have to re-import all files, so your first project startup might take a little longer)

## Additional information

- `[ ]` Changes UI
- `[ ]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:

### CODE REVIEW

File: docs/methods/Actor.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+Makes the next non-action line spoken by the provided actor but hides the name.
+
+Examples: 
+  - `&SPEAK_UNKNOWN:Arin`
+
+## NARRATE
+
+Instant
+
+Makes the next non-action line spoken by a "narrator" actor.
```
Comment (PhoebeMitchell) [inline docs/methods/Actor.md:53]: Maybe not entirely relevant to this PR bu these descriptions are a bit ambiguous. "Hiding the name" could be interpreted as hiding the name box which is what Narrate does, and "spoken by a narrator actor" doesn't convey that this action hides the name box.

File: docs_generator/Properties/launchSettings.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "profiles": {
+    "scanner": {
+      "commandName": "Project",
+      "workingDirectory": "../unity"
```
Comment (PhoebeMitchell) [inline docs_generator/Properties/launchSettings.json]: Would it be possible to rename /unity to something more specific? So it's more identifiable in Unity Hub

File: unity/Assets/ScriptableObjects/Actors/JudgeBrent.asset [hunk 3]
```diff
@@ -10,7 +10,7 @@ MonoBehaviour:
   m_Enabled: 1
   m_EditorHideFlags: 0
   m_Script: {fileID: 11500000, guid: dfc981f29381844a991ebe930492451f, type: 3}
-  m_Name: BrentJudge
+  m_Name: JudgeBrent
```
Comment (PhoebeMitchell) [inline unity/Assets/ScriptableObjects/Actors/JudgeBrent.asset]: Need to update the cross examination tests, they're still looking looking for BrentJudge

**Review by PhoebeMitchell — COMMENTED** [review]
Admittedly a lot of this out of my understanding, but the project appears to open and run fine, and I have some comments and some tests that need fixing.

File: docs_generator/Properties/launchSettings.json (hunk 2 above)
Comment (ViMaSter) [reply docs_generator/Properties/launchSettings.json]: Hm... Fair point. Just noticed that Unity always uses the folder name to deduct the **project** name and ignores any **product** names given in the Player settings. I go by path out of habit, (`C:\prj\6_justice\0_git` but also don't have that many Unity projects on my machine. : D

Anything specific you have in mind? I do find it redundant to have both the repo folder and sub folder named something project-specific, but I totally see the point.

Would `unity-ggjj` work for you?

File: docs/methods/Actor.md (hunk 1 above)
Comment (ViMaSter) [reply docs/methods/Actor.md:53]: How about these alternatives?

`NARRATE`: Makes the next non-action line appear without the actor name box on top, to imply a "narrator"
`SPEAK_UNKNOWN`: Makes the next non-action appear spoken, but obscures the name with `???` until introduced with `SPEAK`

File: docs_generator/Properties/launchSettings.json (hunk 2 above)
Comment (PhoebeMitchell) [reply docs_generator/Properties/launchSettings.json]: unity-ggjj is fine. Though I don't expect I will forget what "unity" is any time soon, still like to have to have some kind of indicator 👍

**Comment by ViMaSter** [comment]
[Moved this to a PR with a branch of inside this repository](https://github.com/Studio-Lovelies/GG-JointJustice-Unity/pull/152), as the GitHub Actions can't properly run on this repository, if they're inside another repo (as it's attempting to push docs on this repo, which it can't, because it's on my `ViMaSter/`-organization.)

**Pull request closed without merging**